folksonomy/models.py

```diff
@@ -24,21 +24,23 @@ class ProductTag(BaseModel):
 
     @validator('product')
     def product_check(cls, v):
+        if not v:
+            raise ValueError('barcode cannot be empty')
         if not re.fullmatch(re_barcode, v):
-            raise ValueError('product is limited to 13 digits')
+            raise ValueError('barcode must be a number')
```

Comment:
No, a code with digits is not always a "number". 00123 is not a "number" but it can be a true barcode.

Reply:
Maybe for more clarity, I could give the error "Barcode should contain only digits of 0-9". Does this sound good @CharlesNepote?